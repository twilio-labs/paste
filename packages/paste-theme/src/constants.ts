import {ValueOf} from '@twilio-paste/types';

export const ThemeVariants = {
  DEFAULT: 'default',
  CONSOLE: 'console',
  SENDGRID: 'sendgrid',
  FLEX: 'flex',
} as const;

// eslint-disable-next-line @typescript-eslint/no-redeclare
export type ThemeVariants = ValueOf<typeof ThemeVariants>;
