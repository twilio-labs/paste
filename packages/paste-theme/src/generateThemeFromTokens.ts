import {
  backgroundColors as BackgroundColors,
  borderColors as BorderColors,
  borderWidths as BorderWidths,
  radii as Radii,
  fonts as Fonts,
  fontSizes as FontSizes,
  fontWeights as FontWeights,
  lineHeights as LineHeights,
  boxShadows as BoxShadows,
  sizings as Sizings,
  spacings as Spacings,
  textColors as TextColors,
  zIndices as ZIndices,
} from '@twilio-paste/design-tokens';

import {GenericThemeShape} from './types/GenericThemeShape';

export const generateThemeFromTokens = (
  backgroundColors: Partial<{[key in keyof typeof BackgroundColors]: any}>,
  borderColors: Partial<{[key in keyof typeof BorderColors]: any}>,
  borderWidths: Partial<{[key in keyof typeof BorderWidths]: any}>,
  radii: Partial<{[key in keyof typeof Radii]: any}>,
  fonts: Partial<{[key in keyof typeof Fonts]: any}>,
  fontSizes: Partial<{[key in keyof typeof FontSizes]: any}>,
  fontWeights: Partial<{[key in keyof typeof FontWeights]: any}>,
  lineHeights: Partial<{[key in keyof typeof LineHeights]: any}>,
  boxShadows: Partial<{[key in keyof typeof BoxShadows]: any}>,
  // there are some sizes we expect must appear to generate breakpoints and icons sizes
  sizings: {[key in keyof typeof Sizings]: any},
  spacings: Partial<{[key in keyof typeof Spacings]: any}>,
  textColors: Partial<{[key in keyof typeof TextColors]: any}>,
  zIndices: Partial<{[key in keyof typeof ZIndices]: any}>
): GenericThemeShape => {
  // default breakpoints
  const breakpoints = [sizings.size40, sizings.size100, sizings.size120];

  return {
    shadows: boxShadows,
    borderWidths,
    radii,
    breakpoints,
    textColors,
    borderColors,
    backgroundColors,
    fonts,
    fontSizes,
    fontWeights,
    lineHeights,
    widths: sizings,
    maxWidths: sizings,
    minWidths: sizings,
    heights: sizings,
    maxHeights: sizings,
    minHeights: sizings,
    sizes: sizings,
    iconSizes: {
      sizeIcon10: sizings.sizeIcon10,
      sizeIcon20: sizings.sizeIcon20,
      sizeIcon30: sizings.sizeIcon30,
      sizeIcon40: sizings.sizeIcon40,
      sizeIcon50: sizings.sizeIcon50,
      sizeIcon60: sizings.sizeIcon60,
      sizeIcon70: sizings.sizeIcon70,
      sizeIcon80: sizings.sizeIcon80,
      sizeIcon90: sizings.sizeIcon90,
      sizeIcon100: sizings.sizeIcon100,
      sizeIcon110: sizings.sizeIcon110,
    },
    space: spacings,
    zIndices,
  };
};
