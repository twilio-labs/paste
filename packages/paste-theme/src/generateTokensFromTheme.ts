import {GenericThemeShape} from './types/GenericThemeShape';
import {GenericTokenShape} from './types/GenericTokenShape';

export const generateTokensFromTheme = (
  backgroundColors: GenericThemeShape['backgroundColors'],
  borderColors: GenericThemeShape['borderColors'],
  borderWidths: GenericThemeShape['borderWidths'],
  radii: GenericThemeShape['radii'],
  fonts: GenericThemeShape['fonts'],
  fontSizes: GenericThemeShape['fontSizes'],
  fontWeights: GenericThemeShape['fontWeights'],
  lineHeights: GenericThemeShape['lineHeights'],
  shadows: GenericThemeShape['shadows'],
  // there are some sizes we expect must appear to generate breakpoints and icons sizes
  sizes: GenericThemeShape['sizes'],
  space: GenericThemeShape['space'],
  textColors: GenericThemeShape['textColors'],
  zIndices: GenericThemeShape['zIndices']
): Partial<GenericTokenShape> => {
  return {
    boxShadows: shadows,
    borderWidths,
    radii,
    textColors,
    borderColors,
    backgroundColors,
    fonts,
    fontSizes,
    fontWeights,
    lineHeights,
    sizings: sizes,
    spacings: space,
    zIndices,
  };
};
