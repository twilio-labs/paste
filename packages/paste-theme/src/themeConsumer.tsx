import * as React from 'react';
import {ThemeContext} from '@emotion/core';
import {ThemeShape} from '@twilio-paste/theme-tokens';

export interface ThemeConsumerProps {
  children: ({theme}: {theme: ThemeShape}) => React.ReactNode;
}
const ThemeConsumer = ({children, ...props}: ThemeConsumerProps): React.ReactElement => {
  if (children == null || typeof children !== 'function') {
    throw new Error('[ThemeConsumer]: You must pass a function as children');
  }
  return (
    <ThemeContext.Consumer>
      {theme => {
        const newTheme = theme as ThemeShape;
        return children({...props, theme: newTheme});
      }}
    </ThemeContext.Consumer>
  );
};

export {ThemeConsumer};
