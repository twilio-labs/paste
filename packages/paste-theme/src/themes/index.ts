import {DefaultTheme} from './default';
import {SendGridTheme} from './sendgrid';
import {ConsoleTheme} from './console';

export {DefaultTheme, ConsoleTheme, SendGridTheme};
