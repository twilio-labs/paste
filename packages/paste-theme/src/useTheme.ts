import * as React from 'react';
import {ThemeContext} from '@emotion/core';
import {ThemeShape} from '@twilio-paste/theme-tokens';

const useTheme = (): ThemeShape => {
  const context = React.useContext(ThemeContext) as ThemeShape;
  if (!context) {
    throw new Error('[useHook]: must be used within the @twilio-paste/theme provider');
  }
  return context;
};

export {useTheme};
