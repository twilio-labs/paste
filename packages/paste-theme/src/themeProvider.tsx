import * as React from 'react';
import styled from '@emotion/styled';
import {Global, css, SerializedStyles} from '@emotion/core';
import {themeGet} from '@styled-system/theme-get';
import {ThemeProvider as StyledThemeProvider} from 'emotion-theming';
import {DefaultTheme, SendGridTheme, ConsoleTheme} from '@twilio-paste/theme-tokens';
import {ThemeVariants} from './constants';

interface GlobalStyleProps {
  theme: {};
}
const pasteGlobalStyles = (props: GlobalStyleProps): SerializedStyles => css`
  html {
    font-size: 100%;
  }
  body {
    margin: ${themeGet('space.space0')(props)};
    font-size: ${themeGet('fontSizes.fontSize30')(props)};
  }
`;

export const StyledBase = styled.div`
  font-family: ${themeGet('fonts.fontFamilyText')};
  font-size: ${themeGet('fontSizes.fontSize30')};
  line-height: ${themeGet('lineHeights.lineHeight30')};
  color: ${themeGet('textColors.colorText')};
  font-weight: ${themeGet('fontWeights.fontWeightNormal')};

  *,
  *::after,
  *::before {
    box-sizing: border-box;
  }
`;

export interface ThemeProviderProps {
  customBreakpoints?: string[];
  theme?: ThemeVariants;
}

const ThemeProvider: React.FunctionComponent<ThemeProviderProps> = ({
  customBreakpoints,
  theme = ThemeVariants.DEFAULT,
  ...props
}) => {
  let breakpoints = {};
  let themeObject = {};

  switch (theme) {
    case ThemeVariants.SENDGRID:
      themeObject = SendGridTheme;
      breakpoints = customBreakpoints || SendGridTheme.breakpoints;
      break;
    case ThemeVariants.CONSOLE:
      themeObject = ConsoleTheme;
      breakpoints = customBreakpoints || ConsoleTheme.breakpoints;
      break;
    case ThemeVariants.FLEX:
    case ThemeVariants.DEFAULT:
    default:
      themeObject = DefaultTheme;
      breakpoints = customBreakpoints || DefaultTheme.breakpoints;
      break;
  }

  const providerThemeProp = {
    ...themeObject,
    breakpoints,
  };

  return (
    <StyledThemeProvider theme={providerThemeProp}>
      <Global styles={pasteGlobalStyles({theme: providerThemeProp})} />
      <StyledBase {...props} />
    </StyledThemeProvider>
  );
};

export {ThemeProvider};
