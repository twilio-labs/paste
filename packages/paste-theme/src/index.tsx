import {ThemeProvider} from './themeProvider';
import {ThemeConsumer} from './themeConsumer';
import {withTheme} from './withTheme';
import {useTheme} from './useTheme';
import {ThemeVariants} from './constants';

const Theme = {
  Provider: ThemeProvider,
  Consumer: ThemeConsumer,
};

export {ThemeShape, SendGridThemeShape, ConsoleThemeShape} from './types';
export {DefaultTheme, ConsoleTheme, SendGridTheme} from './themes';
export {Theme, withTheme, useTheme, ThemeVariants};
