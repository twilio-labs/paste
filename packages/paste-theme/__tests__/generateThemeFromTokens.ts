import {generateThemeFromTokens} from '../src/generateThemeFromTokens';

describe('generateThemeFromTokens', () => {
  it('should generate a theme in the desired shape based on design tokens provided', () => {
    expect(
      generateThemeFromTokens(
        {colorBackground: 'red'},
        {colorBorder: 'blue'},
        {borderWidth0: '10px'},
        {borderRadius0: '20px'},
        {fontFamilyCode: 'arial'},
        {fontSize100: '300px'},
        {fontWeightBold: '800'},
        {lineHeight0: '20px'},
        {shadow: 'grey'},
        {
          size0: '1px',
          size10: '1px',
          size20: '1px',
          size30: '1px',
          size40: '1px',
          size50: '1px',
          size60: '1px',
          size70: '1px',
          size80: '1px',
          size90: '1px',
          size100: '1px',
          size110: '1px',
          size120: '1px',
          sizeSquare0: '1px',
          sizeSquare10: '1px',
          sizeSquare20: '1px',
          sizeSquare25: '1px',
          sizeSquare30: '1px',
          sizeSquare40: '1px',
          sizeSquare50: '1px',
          sizeSquare60: '1px',
          sizeSquare70: '1px',
          sizeSquare80: '1px',
          sizeSquare90: '1px',
          sizeSquare100: '1px',
          sizeSquare110: '1px',
          sizeSquare120: '1px',
          sizeSquare130: '1px',
          sizeSquare140: '1px',
          sizeSquare150: '1px',
          sizeSquare160: '1px',
          sizeSquare170: '1px',
          sizeSquare180: '1px',
          sizeSquare190: '1px',
          sizeSquare200: '1px',
          sizeIcon10: '1px',
          sizeIcon20: '1px',
          sizeIcon30: '1px',
          sizeIcon40: '1px',
          sizeIcon50: '1px',
          sizeIcon60: '1px',
          sizeIcon70: '1px',
          sizeIcon80: '1px',
          sizeIcon90: '1px',
          sizeIcon100: '1px',
          sizeIcon110: '1px',
          sizeSidebar: '1px',
        },
        {space10: '28px'},
        {colorText: 'black'},
        {zIndex0: '-20'}
      )
    ).toMatchSnapshot();
  });
});
