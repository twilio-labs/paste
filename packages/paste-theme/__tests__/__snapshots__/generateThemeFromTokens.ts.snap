// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateThemeFromTokens should generate a theme in the desired shape based on design tokens provided 1`] = `
Object {
  "backgroundColors": Object {
    "colorBackground": "red",
  },
  "borderColors": Object {
    "colorBorder": "blue",
  },
  "borderWidths": Object {
    "borderWidth0": "10px",
  },
  "breakpoints": Array [
    "1px",
    "1px",
    "1px",
  ],
  "fontSizes": Object {
    "fontSize100": "300px",
  },
  "fontWeights": Object {
    "fontWeightBold": "800",
  },
  "fonts": Object {
    "fontFamilyCode": "arial",
  },
  "heights": Object {
    "size0": "1px",
    "size10": "1px",
    "size100": "1px",
    "size110": "1px",
    "size120": "1px",
    "size20": "1px",
    "size30": "1px",
    "size40": "1px",
    "size50": "1px",
    "size60": "1px",
    "size70": "1px",
    "size80": "1px",
    "size90": "1px",
    "sizeIcon10": "1px",
    "sizeIcon100": "1px",
    "sizeIcon110": "1px",
    "sizeIcon20": "1px",
    "sizeIcon30": "1px",
    "sizeIcon40": "1px",
    "sizeIcon50": "1px",
    "sizeIcon60": "1px",
    "sizeIcon70": "1px",
    "sizeIcon80": "1px",
    "sizeIcon90": "1px",
    "sizeSidebar": "1px",
    "sizeSquare0": "1px",
    "sizeSquare10": "1px",
    "sizeSquare100": "1px",
    "sizeSquare110": "1px",
    "sizeSquare120": "1px",
    "sizeSquare130": "1px",
    "sizeSquare140": "1px",
    "sizeSquare150": "1px",
    "sizeSquare160": "1px",
    "sizeSquare170": "1px",
    "sizeSquare180": "1px",
    "sizeSquare190": "1px",
    "sizeSquare20": "1px",
    "sizeSquare200": "1px",
    "sizeSquare25": "1px",
    "sizeSquare30": "1px",
    "sizeSquare40": "1px",
    "sizeSquare50": "1px",
    "sizeSquare60": "1px",
    "sizeSquare70": "1px",
    "sizeSquare80": "1px",
    "sizeSquare90": "1px",
  },
  "iconSizes": Object {
    "sizeIcon10": "1px",
    "sizeIcon100": "1px",
    "sizeIcon110": "1px",
    "sizeIcon20": "1px",
    "sizeIcon30": "1px",
    "sizeIcon40": "1px",
    "sizeIcon50": "1px",
    "sizeIcon60": "1px",
    "sizeIcon70": "1px",
    "sizeIcon80": "1px",
    "sizeIcon90": "1px",
  },
  "lineHeights": Object {
    "lineHeight0": "20px",
  },
  "maxHeights": Object {
    "size0": "1px",
    "size10": "1px",
    "size100": "1px",
    "size110": "1px",
    "size120": "1px",
    "size20": "1px",
    "size30": "1px",
    "size40": "1px",
    "size50": "1px",
    "size60": "1px",
    "size70": "1px",
    "size80": "1px",
    "size90": "1px",
    "sizeIcon10": "1px",
    "sizeIcon100": "1px",
    "sizeIcon110": "1px",
    "sizeIcon20": "1px",
    "sizeIcon30": "1px",
    "sizeIcon40": "1px",
    "sizeIcon50": "1px",
    "sizeIcon60": "1px",
    "sizeIcon70": "1px",
    "sizeIcon80": "1px",
    "sizeIcon90": "1px",
    "sizeSidebar": "1px",
    "sizeSquare0": "1px",
    "sizeSquare10": "1px",
    "sizeSquare100": "1px",
    "sizeSquare110": "1px",
    "sizeSquare120": "1px",
    "sizeSquare130": "1px",
    "sizeSquare140": "1px",
    "sizeSquare150": "1px",
    "sizeSquare160": "1px",
    "sizeSquare170": "1px",
    "sizeSquare180": "1px",
    "sizeSquare190": "1px",
    "sizeSquare20": "1px",
    "sizeSquare200": "1px",
    "sizeSquare25": "1px",
    "sizeSquare30": "1px",
    "sizeSquare40": "1px",
    "sizeSquare50": "1px",
    "sizeSquare60": "1px",
    "sizeSquare70": "1px",
    "sizeSquare80": "1px",
    "sizeSquare90": "1px",
  },
  "maxWidths": Object {
    "size0": "1px",
    "size10": "1px",
    "size100": "1px",
    "size110": "1px",
    "size120": "1px",
    "size20": "1px",
    "size30": "1px",
    "size40": "1px",
    "size50": "1px",
    "size60": "1px",
    "size70": "1px",
    "size80": "1px",
    "size90": "1px",
    "sizeIcon10": "1px",
    "sizeIcon100": "1px",
    "sizeIcon110": "1px",
    "sizeIcon20": "1px",
    "sizeIcon30": "1px",
    "sizeIcon40": "1px",
    "sizeIcon50": "1px",
    "sizeIcon60": "1px",
    "sizeIcon70": "1px",
    "sizeIcon80": "1px",
    "sizeIcon90": "1px",
    "sizeSidebar": "1px",
    "sizeSquare0": "1px",
    "sizeSquare10": "1px",
    "sizeSquare100": "1px",
    "sizeSquare110": "1px",
    "sizeSquare120": "1px",
    "sizeSquare130": "1px",
    "sizeSquare140": "1px",
    "sizeSquare150": "1px",
    "sizeSquare160": "1px",
    "sizeSquare170": "1px",
    "sizeSquare180": "1px",
    "sizeSquare190": "1px",
    "sizeSquare20": "1px",
    "sizeSquare200": "1px",
    "sizeSquare25": "1px",
    "sizeSquare30": "1px",
    "sizeSquare40": "1px",
    "sizeSquare50": "1px",
    "sizeSquare60": "1px",
    "sizeSquare70": "1px",
    "sizeSquare80": "1px",
    "sizeSquare90": "1px",
  },
  "minHeights": Object {
    "size0": "1px",
    "size10": "1px",
    "size100": "1px",
    "size110": "1px",
    "size120": "1px",
    "size20": "1px",
    "size30": "1px",
    "size40": "1px",
    "size50": "1px",
    "size60": "1px",
    "size70": "1px",
    "size80": "1px",
    "size90": "1px",
    "sizeIcon10": "1px",
    "sizeIcon100": "1px",
    "sizeIcon110": "1px",
    "sizeIcon20": "1px",
    "sizeIcon30": "1px",
    "sizeIcon40": "1px",
    "sizeIcon50": "1px",
    "sizeIcon60": "1px",
    "sizeIcon70": "1px",
    "sizeIcon80": "1px",
    "sizeIcon90": "1px",
    "sizeSidebar": "1px",
    "sizeSquare0": "1px",
    "sizeSquare10": "1px",
    "sizeSquare100": "1px",
    "sizeSquare110": "1px",
    "sizeSquare120": "1px",
    "sizeSquare130": "1px",
    "sizeSquare140": "1px",
    "sizeSquare150": "1px",
    "sizeSquare160": "1px",
    "sizeSquare170": "1px",
    "sizeSquare180": "1px",
    "sizeSquare190": "1px",
    "sizeSquare20": "1px",
    "sizeSquare200": "1px",
    "sizeSquare25": "1px",
    "sizeSquare30": "1px",
    "sizeSquare40": "1px",
    "sizeSquare50": "1px",
    "sizeSquare60": "1px",
    "sizeSquare70": "1px",
    "sizeSquare80": "1px",
    "sizeSquare90": "1px",
  },
  "minWidths": Object {
    "size0": "1px",
    "size10": "1px",
    "size100": "1px",
    "size110": "1px",
    "size120": "1px",
    "size20": "1px",
    "size30": "1px",
    "size40": "1px",
    "size50": "1px",
    "size60": "1px",
    "size70": "1px",
    "size80": "1px",
    "size90": "1px",
    "sizeIcon10": "1px",
    "sizeIcon100": "1px",
    "sizeIcon110": "1px",
    "sizeIcon20": "1px",
    "sizeIcon30": "1px",
    "sizeIcon40": "1px",
    "sizeIcon50": "1px",
    "sizeIcon60": "1px",
    "sizeIcon70": "1px",
    "sizeIcon80": "1px",
    "sizeIcon90": "1px",
    "sizeSidebar": "1px",
    "sizeSquare0": "1px",
    "sizeSquare10": "1px",
    "sizeSquare100": "1px",
    "sizeSquare110": "1px",
    "sizeSquare120": "1px",
    "sizeSquare130": "1px",
    "sizeSquare140": "1px",
    "sizeSquare150": "1px",
    "sizeSquare160": "1px",
    "sizeSquare170": "1px",
    "sizeSquare180": "1px",
    "sizeSquare190": "1px",
    "sizeSquare20": "1px",
    "sizeSquare200": "1px",
    "sizeSquare25": "1px",
    "sizeSquare30": "1px",
    "sizeSquare40": "1px",
    "sizeSquare50": "1px",
    "sizeSquare60": "1px",
    "sizeSquare70": "1px",
    "sizeSquare80": "1px",
    "sizeSquare90": "1px",
  },
  "radii": Object {
    "borderRadius0": "20px",
  },
  "shadows": Object {
    "shadow": "grey",
  },
  "sizes": Object {
    "size0": "1px",
    "size10": "1px",
    "size100": "1px",
    "size110": "1px",
    "size120": "1px",
    "size20": "1px",
    "size30": "1px",
    "size40": "1px",
    "size50": "1px",
    "size60": "1px",
    "size70": "1px",
    "size80": "1px",
    "size90": "1px",
    "sizeIcon10": "1px",
    "sizeIcon100": "1px",
    "sizeIcon110": "1px",
    "sizeIcon20": "1px",
    "sizeIcon30": "1px",
    "sizeIcon40": "1px",
    "sizeIcon50": "1px",
    "sizeIcon60": "1px",
    "sizeIcon70": "1px",
    "sizeIcon80": "1px",
    "sizeIcon90": "1px",
    "sizeSidebar": "1px",
    "sizeSquare0": "1px",
    "sizeSquare10": "1px",
    "sizeSquare100": "1px",
    "sizeSquare110": "1px",
    "sizeSquare120": "1px",
    "sizeSquare130": "1px",
    "sizeSquare140": "1px",
    "sizeSquare150": "1px",
    "sizeSquare160": "1px",
    "sizeSquare170": "1px",
    "sizeSquare180": "1px",
    "sizeSquare190": "1px",
    "sizeSquare20": "1px",
    "sizeSquare200": "1px",
    "sizeSquare25": "1px",
    "sizeSquare30": "1px",
    "sizeSquare40": "1px",
    "sizeSquare50": "1px",
    "sizeSquare60": "1px",
    "sizeSquare70": "1px",
    "sizeSquare80": "1px",
    "sizeSquare90": "1px",
  },
  "space": Object {
    "space10": "28px",
  },
  "textColors": Object {
    "colorText": "black",
  },
  "widths": Object {
    "size0": "1px",
    "size10": "1px",
    "size100": "1px",
    "size110": "1px",
    "size120": "1px",
    "size20": "1px",
    "size30": "1px",
    "size40": "1px",
    "size50": "1px",
    "size60": "1px",
    "size70": "1px",
    "size80": "1px",
    "size90": "1px",
    "sizeIcon10": "1px",
    "sizeIcon100": "1px",
    "sizeIcon110": "1px",
    "sizeIcon20": "1px",
    "sizeIcon30": "1px",
    "sizeIcon40": "1px",
    "sizeIcon50": "1px",
    "sizeIcon60": "1px",
    "sizeIcon70": "1px",
    "sizeIcon80": "1px",
    "sizeIcon90": "1px",
    "sizeSidebar": "1px",
    "sizeSquare0": "1px",
    "sizeSquare10": "1px",
    "sizeSquare100": "1px",
    "sizeSquare110": "1px",
    "sizeSquare120": "1px",
    "sizeSquare130": "1px",
    "sizeSquare140": "1px",
    "sizeSquare150": "1px",
    "sizeSquare160": "1px",
    "sizeSquare170": "1px",
    "sizeSquare180": "1px",
    "sizeSquare190": "1px",
    "sizeSquare20": "1px",
    "sizeSquare200": "1px",
    "sizeSquare25": "1px",
    "sizeSquare30": "1px",
    "sizeSquare40": "1px",
    "sizeSquare50": "1px",
    "sizeSquare60": "1px",
    "sizeSquare70": "1px",
    "sizeSquare80": "1px",
    "sizeSquare90": "1px",
  },
  "zIndices": Object {
    "zIndex0": "-20",
  },
}
`;
